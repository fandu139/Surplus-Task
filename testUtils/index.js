import AuthContext from '../App/context/Auth';
import React from 'react';
import { act, fireEvent, render } from '@testing-library/react-native';

const customRender = (ui, options) => {
  const isAuthenticated = options?.isAuthenticated || false;
  const renderOptions = options?.rendererOptions;
  const authContext = options?.authContext;
  return render(
    <AuthContext.Provider value={{ isAuthenticated, ...authContext }}>{ui}</AuthContext.Provider>,
    renderOptions,
  );
};

// override render method
export { customRender as render };

export function wait(ms) {
  return act(() => new Promise((r) => setTimeout(r, ms)));
}

// Due to debounce on Button UI Kit, we need to add milisecond waiting timer
export function fireAsyncPress(element, ms = 100) {
  fireEvent.press(element);
  return wait(ms);
}
