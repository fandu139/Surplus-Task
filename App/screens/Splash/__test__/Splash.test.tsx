import AppSplashScreen from '..';
import { render } from '../../../../testUtils';
import React from 'react';

describe('splash screen test', () => {
  describe('snapshot test', () => {
    it('should render correctly', () => {
      const tree = render(<AppSplashScreen />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
