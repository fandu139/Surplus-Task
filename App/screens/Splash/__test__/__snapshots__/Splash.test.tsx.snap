// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`splash screen test snapshot test should render correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#d4f6f3",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      [
        {
          "overflow": "hidden",
        },
        {
          "height": 200,
          "width": 200,
        },
      ]
    }
  >
    <FastImageView
      defaultSource={null}
      resizeMode="contain"
      source={
        {
          "testUri": "../../../App/assets/images/surplus_logo.png",
        }
      }
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
  <Text
    style={
      {
        "textAlign": "center",
        "width": 250,
      }
    }
  >
    Membantu dan mendukung masyarakat dalam memahami dan menerapkan pola produksi dan konsumsi pangan yang selaras dengan keberlanjutan sosial dan lingkungan
  </Text>
  <View
    style={
      {
        "bottom": 100,
        "position": "absolute",
      }
    }
  >
    <ActivityIndicator
      accessibilityLabel="loading-indicator"
      color="#47bba5"
      size="large"
      testID="loading-indicator"
    />
  </View>
</View>
`;
